File: DyeHard/Platform.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using XNACS1Lib;
using Microsoft.Xna.Framework;

namespace Dyehard
{
    class Platform
    {
        private const int SEGMENT_COUNT = 20;
        public static float height = 1.2f;
        private Hero hero;
        private List<Enemy> enemies;
        private List<Obstacle> obstacles;

        public Platform(int offset, Hero hero, List<Enemy> enemies, float leftEdge, bool continuous = false)
        {
            this.hero = hero;
            this.enemies = enemies;
            this.obstacles = new List<Obstacle>();
            // set up platform
;
            float Ypos = ((offset * 1f) / Stargate.GATE_COUNT) * Game.topEdge();


            if (continuous)
            {
                float Xpos = leftEdge + (Stargate.width / 2);
                Obstacle obstacle = new Obstacle(hero, enemies, new Vector2(Xpos, Ypos), Stargate.width, height);
                obstacles.Add(obstacle);
            }
            else
            {
                // randomly fill platform
                float width = Stargate.width / SEGMENT_COUNT;
                Obstacle obstacle;
                int consecutiveChance = 10;
                bool platform = true;
                for (int i = 0; i < SEGMENT_COUNT; i++)
                {

                    if (platform)
                    {
                        float Xpos = (width * 0.5f) + leftEdge + (i * width);
                        obstacle = new Obstacle(hero, enemies, new Vector2(Xpos, Ypos), width, height);
                        obstacles.Add(obstacle);
                    }

                    consecutiveChance -= 2;
                    if (XNACS1Base.RandomInt(consecutiveChance) == 0) {
                        platform = !platform;
                        consecutiveChance = 10;    
                    }   
                }
            }
        }

        public void move()
        {
            foreach (Obstacle obstacle in obstacles)
            {
                obstacle.move();
            }
        }

        public void draw()
        {
            foreach (Obstacle obstacle in obstacles)
            {
                obstacle.draw();
            }
        }

        public void interact()
        {
            foreach (Obstacle obstacle in obstacles)
            {
                obstacle.interact();
            }
        }
    }
}

File: Java/DyeHardUser/src/Platform.java


import java.util.Random;

import dyehard.World.PlatformSingle;
import Engine.BaseCode;
import Engine.Vector2;

public class Platform {
    public static float height = 1.25f;
    private final float width = 6.5f;
    private static Random RANDOM = new Random();

    public Platform(int offset, float leftEdge, boolean continuous) {
        fillPlatform(offset, leftEdge, continuous);
    }

    private void fillPlatform(int offset, float leftEdge, boolean continuous) {
        // set up platform
        float Ypos = ((offset * 1f) / Stargate.GATE_COUNT)
                * BaseCode.world.getHeight();
        int numPlat = (int) (Stargate.WIDTH / width);
        if (continuous) {
            for (int i = 0; i < numPlat; i++) {
                float Xpos = (width * i) + leftEdge + (width / 2);
                new PlatformSingle(new Vector2(Xpos, Ypos));
            }
        } else {
            // randomly fill platform
            int consecutiveChance = 10;
            boolean platform = true;
            for (int i = 0; i < numPlat; i++) {
                if (consecutiveChance <= 0
                        || RANDOM.nextInt(consecutiveChance) == 0) {
                    platform = !platform;
                    consecutiveChance = 10;
                }
                if (platform) {
                    float Xpos = (width * i) + leftEdge + (width / 2);
                    new PlatformSingle(new Vector2(Xpos, Ypos));
                }
                consecutiveChance -= 1;
            }
        }
    }
}
